src/main/java/com/deavensoft/paketomat/center/CenterServiceImpl.java

```diff
@@ -54,6 +62,49 @@ public void updateStatus(Long code, Status status) {
             }
         }
     }
+    public void payment(Long id, Paid paid)
+    {
+        List<Package> packages = centerRepository.findAll();
+        for (Package p : packages) {
+            if(p.getId().equals(id)){
+                p.setPaid(paid);
+                sendMailToUser(p.getUser().getEmail(),paid);
+                centerRepository.save(p);
+            }
+        }
+
+
+    }
+
+    public void sendMailToUser(String email, Paid p) {
+        EmailDetails emailSender = new EmailDetails();
+        emailSender.setRecipient(email);
+        if (Paid.PAID == p) {
+            emailSender.setMsgBody("Your package is in the paketomat and is ready to be picked up and the code is" +" "+
+                    generateCode());
+        } else if (Paid.NOT_PAID == p) {
+            emailSender.setMsgBody("Your package is in the paketomat and is ready to be paid");
+        } else if (Paid.UN_SUCESSFULL == p) {
+            emailSender.setMsgBody("Your package is in the paketomat, the payment was unsuccesfull, try again to pay for the package");
+        }
+        emailSender.setAttachment("");
+        emailSender.setSubject("Package arrived in the paketomat");
+        Map<String, Object> model = new HashMap<>();
+        model.put("msgBody", emailSender.getMsgBody());
+        emailService.sendMailWithTemplate(emailSender, model);
+        log.info("E-Mail is sent to the end user");
+
+    }
+
+    public String generateCode() {
+        SecureRandom random = new SecureRandom();
+        int num = random.nextInt(10000);
```

Comment:
**A magic number** is a numeric value that’s encountered in the source but has no obvious meaning. Your code uses a number that has a certain meaning to it. This “anti-pattern” makes it harder to understand the program and refactor the code.

**Solution**
Replace this number with a constant that has a human-readable name explaining the meaning of the number.

Reply:
Done.

Reply:
Check it again.

Reply:
?

Reply:
Its fixed